tests/wrappers/vector/test_record_episode_statistics.py

```diff
@@ -56,16 +60,41 @@ def test_record_episode_statistics(num_envs, env_id="CartPole-v1", num_steps=100
         data_equivalence(wrapper_vector_reward, vector_wrapper_reward)
         data_equivalence(wrapper_vector_terminated, vector_wrapper_terminated)
         data_equivalence(wrapper_vector_truncated, vector_wrapper_truncated)
+        data_equivalence(wrapper_vector_info, vector_wrapper_info)
 
-        if "episode" in wrapper_vector_info:
-            assert "episode" in vector_wrapper_info
+        dones[t] = wrapper_vector_terminated | wrapper_vector_truncated
+        rewards[t] = wrapper_vector_reward
+
+        assert np.all(
+            wrapper_vector_info.get("_episode", np.zeros(num_envs, dtype=bool))
+            == dones[t]
+        )
 
+        if "episode" in wrapper_vector_info:
             wrapper_vector_time = wrapper_vector_info["episode"].pop("t")
             vector_wrapper_time = vector_wrapper_info["episode"].pop("t")
             assert wrapper_vector_time.shape == vector_wrapper_time.shape
             assert wrapper_vector_time.dtype == vector_wrapper_time.dtype
 
-        data_equivalence(wrapper_vector_info, vector_wrapper_info)
```

Comment:
Once we pop `t`, why can we not just do `data_equivalence` for the two infos?
Why did this test fail previously?

Reply:
I messed on the testing of this, it should be `assert data_equivalence(...)`

The new testing code should be 
```python
        assert data_equivalence(wrapper_vector_obs, vector_wrapper_obs)
        assert data_equivalence(wrapper_vector_reward, vector_wrapper_reward)
        assert data_equivalence(wrapper_vector_terminated, vector_wrapper_terminated)
        assert data_equivalence(wrapper_vector_truncated, vector_wrapper_truncated)

        if "episode" in wrapper_vector_info:
            assert "episode" in vector_wrapper_info

            wrapper_vector_time_taken = wrapper_vector_info["episode"].pop("t")
            vector_wrapper_time_taken = vector_wrapper_info["episode"].pop("t")

            assert wrapper_vector_time_taken.shape == vector_wrapper_time_taken.shape
            assert wrapper_vector_time_taken.dtype == vector_wrapper_time_taken.dtype

            vector_wrapper_info["episode"].pop("_l")
            vector_wrapper_info["episode"].pop("_r")
            vector_wrapper_info["episode"].pop("_t")

        assert data_equivalence(wrapper_vector_info, vector_wrapper_info)
```
This should now test it correctly.

I added `max_episode_steps` to the `make_vec` such that episode ends for this to actually be tested (update to main, this was previously broken)

Reply:
Oh, that is why it did not fail when I moved the info dict comparison! Do you want me to fix the test, or are you going to do it?

Reply:
Could you update the test to my suggested code above, I think it is simplier than your code and achieves the same thing.

Reply:
I totally agree that your version is simpler. For some reason, I thought the test did not consider the episode lengths at all, but after looking at it more carefully, I now understand what it actually does. I just changed the code accordingly.

Reply:
Yeah, it does that implicitly with the info check at the end